types/index.d.ts

```diff
@@ -618,6 +618,16 @@ declare module 'mongoose' {
    */
   export type UpdateQuery<T> = _UpdateQuery<T> & AnyObject;
 
+  /**
+   * A more strict form of UpdateQuery that enforces updating only
+   * known top-level properties.
+   * @example
+   * ```js
+   * { age: 30 }
+   * ```
+   */
+  export type UpdateQueryKnownOnly<T> = _UpdateQuery<T, {}>;
```

Comment:
it would probably good to have a example of how this type is meant to be used.

is it meant to be used like `.find({} as UpdateQueryKnownOnly<IUser>)` or `.find<UpdateQueryKnownOnly<IUser>({})` or a completely different way?

Reply:
More along the lines of this:

```ts
function updateUser(update: UpdateQueryKnownOnly<IUser>) {
  return User.updateOne({}, update);
}
```

Use `UpdateQueryKnownOnly` for extra strictness in params and definitions when you want it. Does that make sense?

Reply:
so if no conditions are given via parameters and are "inline", the following should be used, right?
`User.updateOne({}, { something: 1 } as UpdateQueryKnownOnly<IUser>)`